.jenkins/cscs/env-clang-11.sh

```diff
@@ -4,30 +4,22 @@
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
```

Comment:
Before merging I'll try to remind myself why I had to change this from clang 13 to clang 11.

Reply:
The reason was clang 12 and 13 not building with newer GCCs, and if an older one is used (like the system GCC which is version 7) the headers are not new enough to use C++17.

I'm retrying the clang 13 builds with newer GCCs to see if things have improved but if not this will have to stay on clang 11 for the moment.

Reply:
There's hope for getting newer clangs working: https://github.com/spack/spack/pull/30583. However, I'll leave that out of this PR.